### ISSUE TITLE
Error in event listener when uploading file in front-end form

### ISSUE THREAD

**Issue by jasonvarga** [statamic/cms#9512]
### Bug description

You receive an error like `The file "/private/var/tmp/phpp4qVZI" does not exist` when:

- You have a form with an `assets` field
- You pick a file to upload
- You submit the front-end form
- You have an event listener listening for `FormSubmitted`
- You try to access a request variable inside the listener

You _must_ choose a file when submitting the form otherwise you don't get the error.

### How to reproduce

```antlers
{{ form:example }}
  {{ fields }}
    {{ display }} {{ field }} <br />
  {{ /fields }}

  <button>submit</button>
{{ /form:example }}

```

```php
Event::listen(function (FormSubmitted $event) {
    $value = request('anything'); // this causes the error.
    ray($value);
});
```

### Environment

```yaml
Statamic 4.48.0
```

### Installation

Fresh statamic/statamic site via CLI

### Antlers Parser

None

### Additional details

This was introduced in 4.38.0 by #8886, but I can't see why.

**Comment by ryanmitchell** [comment]
Its coming from this line: https://github.com/statamic/cms/blob/8ee617d374bd2689ab3a944a2363d71d44deda1a/src/Http/Requests/FrontendFormRequest.php#L113

Might it be getting the file before its actually been processed/uploaded?

**Comment by ryanmitchell** [comment]
oh no its the opposite... the file has been uploaded by that point so calling a request results in it trying to normalise to a file that no longer exists.

**Comment by martyf** [comment]
This is an issue with addons like [aryehraber/statamic-captcha](https://github.com/aryehraber/statamic-captcha) that use a listener.

This listens for a `FormSubmitted` event, and fails with the same error noted above.

Basically this means sites that use this addon (or [stoffelio/statamic-turnstile](https://github.com/stoffelio/statamic-turnstile)) are unable to have captcha. Which is a bit of a big one.

Fix, temporarily, is to rollback to `4.37`, but you lose precognition.

**Comment by ryanmitchell** [comment]
I wonder if some tweaking of the logic here is required:

https://github.com/statamic/cms/blob/76bd9721f848d4a851518f2490b58b7309e9ac31/src/Http/Controllers/FormController.php#L53-L61

At the moment the FormSubmitted is fired after the upload, I guess to allow listeners access to the assets, however by the time the event has fired the temporary asset files have been removed, which is why this error occurs. The way things are set up also means if a FormSubmitted causes a failure the assets are still uploaded, which doesn't seem right to me.

Maybe the FormController needs to:
1. Not delete temporary files until after the submitted event so they are available to request()
2. Be updated to delete any uploaded files when a silent failure or validation error is triggered
